### PR TITLE
Add Github Action for publish spec to NeoFS

### LINKED ISSUE #85: Build and publish to NeoFS
## Is your feature request related to a problem? Please describe.
I'm always frustrated when I can't download a new spec from fs.neo.org or somewhere else.

## Describe the solution you'd like
Reuse https://github.com/nspcc-dev/gh-push-to-neofs to create a workflow building and pushing this spec into some container, it should have some static name and always replace an existing one. This way we'd get a static URL with the spec.

### PR DESCRIPTION
**Pull request by mike-petrov**
Signed-off-by: Mikhail Petrov <mike@nspcc.ru>

closes #85

### CODE REVIEW

File: .github/workflows/neofs.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Publish to NeoFS
+on:
+  push:
+    branches: [ master ]
+
+env:
+  NEOFS_NETWORK_DOMAIN: 'st1.storage.fs.neo.org'
+  NEOFS_HTTP_GATE: 'http.fs.neo.org'
+  PATH_TO_FILES_DIR: 'output/neofs-spec-<revision>.pdf'
```
Comment (roman-khimov) [inline .github/workflows/neofs.yml:9]: Revision?

File: .github/workflows/neofs.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+      - name: Publish to NeoFS
+        uses: nspcc-dev/gh-push-allure-report-to-neofsfs@master
+        with:
+          NEOFS_WALLET: ${{ secrets.NEOFS_WALLET }}
+          NEOFS_WALLET_PASSWORD: ${{ secrets.NEOFS_WALLET_PASSWORD }}
+          NEOFS_NETWORK_DOMAIN: ${{ env.NEOFS_NETWORK_DOMAIN }}
+          NEOFS_HTTP_GATE: ${{ env.NEOFS_HTTP_GATE }}
+          STORE_OBJECTS_CID: ${{ env.STORE_OBJECTS_CID }}
+          PATH_TO_FILES_DIR: ${{ env.REPORT_DIR }}
```
Comment (roman-khimov) [inline .github/workflows/neofs.yml:32]: REPORT_DIR?

File: .github/workflows/neofs.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  NEOFS_HTTP_GATE: 'http.fs.neo.org'
+  STORE_OBJECTS_CID: ''
+
+jobs:
+  push-to-neofs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set PATH_TO_FILES_DIR
+        run: |
+          echo "PATH_TO_FILES_DIR=output/neofs-spec-$(shell git describe --abbrev=8 --dirty --always).pdf"
```
Comment (roman-khimov) [inline .github/workflows/neofs.yml:17]: This better be derived from the Makefile (`make version` of some kind).

**Review by roman-khimov — APPROVED** [review]
Looks legit, @vvarg229?

File: .github/workflows/neofs.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: Publish to NeoFS
+on:
+  push:
+    branches: [ master ]
+
+env:
+  NEOFS_NETWORK_DOMAIN: 'st1.storage.fs.neo.org'
+  NEOFS_HTTP_GATE: 'http.fs.neo.org'
+  STORE_OBJECTS_CID: ''
```
[the remark this answers is not in the document]
Comment (mike-petrov) [reply .github/workflows/neofs.yml:9]: it has to be a specific `STORE_OBJECTS_CID`

I don't know which one to put

[the remark this answers is not in the document]
Comment (carpawell) [reply .github/workflows/neofs.yml:9]: oh, it will be changed later. i see, no problem

[the remark this answers is not in the document]
Comment (roman-khimov) [reply .github/workflows/neofs.yml:9]: It's set in repo vars.

[the remark this answers is not in the document]
Comment (carpawell) [reply .github/workflows/neofs.yml:9]: so the priority is `.yml` files < repo vars?. just to be sure, did not check it

File: .github/workflows/neofs.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+jobs:
+  push-to-neofs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set PATH_TO_FILES_DIR
+        run: |
+          echo "PATH_TO_FILES_DIR=output/neofs-spec-$(make version).pdf"
```
Comment (vvarg229) [inline .github/workflows/neofs.yml:17]: echo "PATH_TO_FILES_DIR=output" >> $GITHUB_ENV

File: .github/workflows/neofs.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+name: Publish to NeoFS
+on:
+  push:
+    branches: [ master ]
+
+env:
```
Comment (vvarg229) [inline .github/workflows/neofs.yml:6]: Remove the variable definitions from here - they will be set by @roman-khimov in the repository settings.

**Review by vvarg229 — CHANGES_REQUESTED** [review]

File: .github/workflows/neofs.yml [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Get pdf file
+        run: make docker/pdf
+
+      - uses: actions/checkout@v4
+      - name: Publish to NeoFS
+        uses: nspcc-dev/gh-push-to-neofs@master
+        with:
+          NEOFS_WALLET: ${{ secrets.NEOFS_WALLET }}
+          NEOFS_WALLET_PASSWORD: ${{ secrets.NEOFS_WALLET_PASSWORD }}
+          NEOFS_NETWORK_DOMAIN: ${{ env.NEOFS_NETWORK_DOMAIN }}
```
Comment (carpawell) [inline .github/workflows/neofs.yml:26]: @vvarg229, why do we have a domain here?

File: .github/workflows/neofs.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches: [ master ]
+
+jobs:
+  push-to-neofs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set PATH_TO_FILES_DIR
+        run: |
+          echo "PATH_TO_FILES_DIR=output" >> $GITHUB_ENV
```
Comment (carpawell) [inline .github/workflows/neofs.yml:12]: why is it hardcoded and not an env? @vvarg229

**Review by carpawell — APPROVED** [review]

File: .github/workflows/neofs.yml (hunk 7 above)
Comment (vvarg229) [reply .github/workflows/neofs.yml:26]: > @vvarg229, why do we have a domain here?

How else do you identify which neofs network to use?

File: .github/workflows/neofs.yml (hunk 8 above)
Comment (vvarg229) [reply .github/workflows/neofs.yml:12]: > why is it hardcoded and not an env? @vvarg229

Yes, you can put this variable into GitHub variables from the web interface

**Review by vvarg229 — DISMISSED** [review]

File: .github/workflows/neofs.yml (hunk 7 above)
Comment (roman-khimov) [reply .github/workflows/neofs.yml:26]: That's about gateway IIRC.

Comment (roman-khimov) [reply .github/workflows/neofs.yml:26]: To produce a link to download/view the uploaded thing.

File: .github/workflows/neofs.yml (hunk 8 above)
Comment (roman-khimov) [reply .github/workflows/neofs.yml:12]: Looks like there is some inconsistency there, we have `PATH_TO_FILES_DIR` in an `env`, but here it's hardcoded to a particular value appropriate for `make docker/pdf`.

File: .github/workflows/neofs.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: actions/checkout@v4
+      - name: Publish to NeoFS
+        uses: nspcc-dev/gh-push-to-neofs@master
+        with:
+          NEOFS_WALLET: ${{ secrets.NEOFS_WALLET }}
+          NEOFS_WALLET_PASSWORD: ${{ secrets.NEOFS_WALLET_PASSWORD }}
+          NEOFS_NETWORK_DOMAIN: ${{ env.NEOFS_NETWORK_DOMAIN }}
+          NEOFS_HTTP_GATE: ${{ env.NEOFS_HTTP_GATE }}
+          STORE_OBJECTS_CID: ${{ env.STORE_OBJECTS_CID }}
+          PATH_TO_FILES_DIR: ${{ env.PATH_TO_FILES_DIR }}
```
Comment (roman-khimov) [inline .github/workflows/neofs.yml:29]: I think we should not have this `env` variable, we know exactly which files we want to upload, so action should receive this output directory and that's it.

File: .github/workflows/neofs.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+      - name: Build Docker image
+        run: make image
+
+      - name: Get pdf file
+        run: make docker/pdf
+
+      - uses: actions/checkout@v4
+      - name: Publish to NeoFS
+        uses: nspcc-dev/gh-push-to-neofs@master
```
Comment (roman-khimov) [inline .github/workflows/neofs.yml:22]: `@master`? I'd use something more stable.

**Review by roman-khimov — CHANGES_REQUESTED** [review]

**Comment by roman-khimov** [comment]
I've added envs/secrets (except PATH_TO_FILES_DIR). But we need to synchronize the name of the uploaded spec with the website then (see nspcc-dev/gh-push-to-neofs#2 also).

File: .github/workflows/neofs.yml (hunk 10 above)
Comment (roman-khimov) [reply .github/workflows/neofs.yml:18]: Please set the latest stable version here.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**